synapse/storage/data_stores/main/cache.py

```diff
@@ -64,7 +81,14 @@ def get_all_updated_caches_txn(txn):
                 LIMIT ?
             """
             txn.execute(sql, (last_id, instance_name, limit))
-            return txn.fetchall()
+            updates = [(row[0], row[1:]) for row in txn]
+            limited = False
+            upto_token = current_id
+            if len(updates) >= limit:
```

Comment:
#7636 seems to use a mixture of `>`, `>=` and `==` to determine if the stream was limited. I'm not sure if this inconsistency is on purpose or not. I would have thought `>` would be the proper choice here to avoid unnecessary work, but maybe that's wrong.

Reply:
Ah, yes, sorry. It mostly depends on the SQL query. Some are `LIMIT ?` and so will only ever return at most `limit` rows. Some have more fuzzy bounds and so could return more. I don't think anywhere uses `>`? If they do it sounds like a bug

Reply:
Ah, `typing` uses `>`, but that is a special case because we have all results in memory and don't query the DB  with a limit

Reply:
Interesting. Is this worth a comment in any of the locations?

Reply:
I've added a comment to the typing one :+1: